### ISSUE TITLE
Examples for SM Distributed DataParallel

### ISSUE THREAD

**Issue by aysusayin** [aws/amazon-sagemaker-examples#1901]
I am trying to write my own training script using Tensorflow SDP. However the examples here are either too complex (Mask RNN - Bert) or too simple(MNIST). The documentation of the library is also not sufficient. Could you add another example? 
And can I use Tensorflow SDP just as I use Horovod?

Thanks!

**Comment by hsl89** [comment]
Hi @aysusayin , what kind of example you want to see that's between too complex and too simple?

**Comment by aysusayin** [comment]
Hi @hsl89 , thanks for the answer.  

I'm interested in implementing my own training script with SDP. I see distributed optimizer in the documentation but I don't clearly understand how and when to use it. I saw that it is used in the BERT example. However, in my opinion, that project is too big to spot where SDP dataparallel is used. 

Also, there are some differences between MNIST example here and the tutorial on the AWS docs (https://docs.aws.amazon.com/sagemaker/latest/dg/data-parallel-modify-sdp.html) which I find confusing.

**Comment by hsl89** [comment]
I agree that it's a little bit funny that the only examples we offer are either bert or something that works on mnist. The script here https://github.com/aws/amazon-sagemaker-examples/blob/master/training/distributed_training/tensorflow/data_parallel/mnist/code/train_tensorflow_smdataparallel_mnist.py can be generalized by changing `mnist_model` to `your_favorite_model`. The logic for data parallel should really not depends on the architecture of the underlying model. At least that's how I see it. 

It would be great if you can say more about why the docs are confusing so that we can iterate on it.

**Comment by hsl89** [comment]
close for lack of activity. pls reopen if you have new input
